deploy/vagrant/core/EnvSetup.rb

```diff
@@ -57,3 +57,45 @@ def config_hosts(name)
     hosts.close unless hosts == nil
   end
 end
+
+require 'yaml'
+require 'open-uri'
+
+def get_tachyon_home
+  @v = YAML.load_file('tachyon_version.yml')
+  
+  type = @v['Type']
+  if type == "Local"
+    puts 'using local tachyon dir'
+    return "../../"
+  end
+
+  home = "../tachyon"
+  if type == "Github"
+    repo = @v['Github']['Repo']
+    hash = @v['Github']['Hash']
+    url = "#{repo}/commit/#{hash}"
+
+    begin
+      open(url)
+    rescue OpenURI::HTTPError
+      $stderr.print "The github URL #{url} is invalid"
+      exit(1)
+    end
+
+    puts "git clone #{url} and mvn package"
+    `if [ -d ../tachyon ]; then rm -rf ../tachyon; fi`
+    `mkdir -p ../tachyon`
+    `pushd ../tachyon > /dev/null; \
+     git init; \
+     git remote add origin #{repo}; \
+     git fetch origin; \
+     git checkout #{hash}; \
+     mvn package -DskipTests; \
```

Comment:
One solution is to provide two options: --test and --hdfs. 

Another solution is, this script only checks if there is a downloaded tachyon directory, if there isn't, download and checkout the commit, else, we assume the user has tested or packaged what they want before running this deployment script, so we remove the logic of packaging from this deployment script to user side.

what do you think?